packages/browser/core-hooks/useDebounce/index.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "react";
+
+const DEFAULT_DELAY = 500;
+
+const useDebounce = <T>(
+  debounceResolutionCallback: (value: T) => void,
+  delay: number = DEFAULT_DELAY
+) => {
+  const [debouncedValue, setDebouncedValue] = useState<T | undefined>(
+    undefined
+  );
+  const updateDebouncedValue = (value: T) => {
+    setDebouncedValue(value);
+  };
```

Comment:
This function is not needed as it offers no additional functionality to the orignal setState